packages/docusaurus-1.x/lib/server/docs.js

```diff
@@ -69,11 +69,19 @@ function mdToHtmlify(oldContent, mdToHtml, metadata, siteConfig) {
     let mdMatch = mdRegex.exec(modifiedLine);
     while (mdMatch !== null) {
       /* Replace it to correct html link */
-      const docsSource = metadata.version
-        ? metadata.source.replace(/version-.*?\//, '')
-        : metadata.source;
+      let docsSource;
+      if (readMetadata.shouldGenerateNextReleaseDocs()) {
+        docsSource = metadata.version
+          ? metadata.source.replace(/version-.*?\//, '')
+          : metadata.source;
+      } else {
+        docsSource = metadata.source;
+      }
       let htmlLink =
-        mdToHtml[resolve(docsSource, mdMatch[1])] || mdToHtml[mdMatch[1]];
+        mdToHtml[resolve(docsSource, mdMatch[1])] ||
+        mdToHtml[mdMatch[1]] ||
+        resolve(docsSource, mdMatch[1]).replace('.md', '.html');
```

Comment:
Looks like this is the cause of test failure, regarding link to non-existing doc. But this helps to resolve versioned docs when `--skip-next-release` build option is used. Is there a better way this can be addressed?

```
 FAIL  packages/docusaurus-1.x/lib/server/__tests__/docs.test.js (6.591s)
  ● mdToHtmlify › transform to correct link

    expect(received).toMatchSnapshot()

    Snapshot name: `mdToHtmlify transform to correct link 1`

    - Snapshot
    + Received

    @@ -4,11 +4,11 @@
      - [doc1](/docs/en/next/doc1)
      - [doc2](/docs/en/next/doc2)

      ### Non-existing Docs

    - - [hahaha](hahaha.md)
    + - [hahaha](hahaha)
```

Reply:
Resolved it by validating whether or not the target file actually exists.